### DISCUSSION TITLE
flutter socketio client always connect failed

### DISCUSSION THREAD

**Discussion by hakanaku1234** [miguelgrinberg/Flask-SocketIO#1597] [Q&A]
hello, I was trying to use flutter socketio client package to connect with flask-socketio, but i can't.

it always show this error message:
`The client is using an unsupported version of the Socket.IO or Engine.IO protocols`

I have no idea how to correct the version between flutter and flask-socketio.

my flask_socketio version is 5.0.1
and there are the flutter client packages i have tried:
https://pub.dev/packages/socket_io_client/versions/0.9.11
https://pub.dev/packages/flutter_socket_io

---

bwt, how can I check the version compatibility of flask-socketio

for example, something like this table:

JavaScript Socket.IO version | Socket.IO protocol revision | Engine.IO protocol revision | python-socketio version | python-engineio version
-- | -- | -- | -- | --
0.9.x | 1, 2 | 1, 2 | Not supported | Not supported
1.x and 2.x | 3, 4 | 3 | 4.x | 3.x
3.x | 5 | 4 | 5.x | 4.x

**Comment by miguelgrinberg** [comment]
I don't use the flutter client, no idea which version of the Socket.IO protocol it implements. You need to find that out. In practical terms if the protocol version 5 did not work for you, then very likely they support version 4, so you can downgrade Flask-SocketIO and its dependencies to the versions that work with version 4 and try again.

**Comment by hakanaku1234** [comment]
Thank you very much!! I will try it out :)

btw, how can I find out `which version of Flask-SocketIO` works with `socket.io protocol version 4`?

**Comment by miguelgrinberg** [comment]
> how can I find out which version of Flask-SocketIO works with socket.io protocol version 4?

The docs for Flask-SocketIO have a version compatibility table.
